device/cuda/src/cca/component_connection.cu

```diff
@@ -764,16 +763,49 @@ component_connection::output_type component_connection::operator()(
         }
     }
 
+    /*
+     * Allocate space on the device for input.
+     */
+    vecmem::unique_alloc_ptr<channel_id[]> channel0_device =
+        vecmem::make_unique_alloc<channel_id[]>(dmr, channel0.size());
+    vecmem::unique_alloc_ptr<channel_id[]> channel1_device =
+        vecmem::make_unique_alloc<channel_id[]>(dmr, channel1.size());
+    vecmem::unique_alloc_ptr<scalar[]> activation_device =
+        vecmem::make_unique_alloc<scalar[]>(dmr, activation.size());
+    vecmem::unique_alloc_ptr<scalar[]> time_device =
+        vecmem::make_unique_alloc<scalar[]>(dmr, time.size());
+    vecmem::unique_alloc_ptr<geometry_id[]> module_id_device =
+        vecmem::make_unique_alloc<geometry_id[]>(dmr, module_id.size());
```

Comment:
Why not `vecmem::data::vector_buffer` objects? Then the code could easily use vector views and device vectors around these things. In sync with how all the other code is written. :thinking: